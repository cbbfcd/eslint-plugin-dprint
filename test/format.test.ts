import { Linter } from "eslint";
import * as parserPlain from "eslint-parser-plain";
import { describe, expect, it } from "vitest";

import DprintIntegration from "../src";

const linter = new Linter();
const linterConfig = {
	parser: "plain",
	rules: {
		dprint: "error",
	},
} satisfies Linter.Config;

linter.defineParser("plain", parserPlain as any);
linter.defineRule("dprint", DprintIntegration.rules.dprint);

describe("format", async () => {
	const fixtureFiles = import.meta.glob("./__fixtures__/format/**", {
		eager: true,
		as: "raw",
	});

	const results = Object.entries(fixtureFiles).map(([file, source]) => {
		const { output } = linter.verifyAndFix(source, linterConfig, file);

		return {
			file,
			output,
		};
	});

	for (const { file, output } of results) {
		it(`File: ${file}`, async () => {
			expect(output).toMatchSnapshot();
		});
	}
});
