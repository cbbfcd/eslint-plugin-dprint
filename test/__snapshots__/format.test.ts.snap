// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`format > File: ./__fixtures__/format/svelte.svelte 1`] = `
"<div></div>
"
`;

exports[`format > File: ./__fixtures__/format/typescript.ts 1`] = `
"const a = "1";
"
`;

exports[`format > File: ./__fixtures__/format/vue.vue 1`] = `
"<template>
  <div></div>
</template>
"
`;
